### PR TITLE
Store a player's default play mode

### PR DESCRIPTION
**Pull request by myssto**
Small change to the `OsuPlayerDataWorker` that stores the player's default play mode when updating ranking information. This is working towards a change in how we display player profiles on web. Using this information, when visiting a player's profile on web we can show the statistics for the target player's preferred mode instead of the logged in user's preferred mode.

### CODE REVIEW

File: API/Entities/Player.cs [hunk 1]
```diff
@@ -41,6 +42,9 @@ public class Player
     [MaxLength(4)]
     public string? Country { get; set; }
 
+    [Column("play_mode")]
+    public OsuEnums.Mode? PlayMode { get; set; }
```
Comment (hburn7) [inline API/Entities/Player.cs:46]: Can we call this `default_ruleset` instead?

File: API/Osu/Multiplayer/OsuApiService.cs [hunk 2]
```diff
@@ -19,6 +19,9 @@ public class OsuApiUser
 
     [JsonProperty("country")]
     public string Country { get; set; } = null!;
+
+    [JsonProperty("play_mode")]
+    public GameMode PlayMode { get; set; }
```
Comment (hburn7) [inline API/Osu/Multiplayer/OsuApiService.cs:24]: Here, would rather refactor `GameMode` to `Ruleset` and `PlayMode` to `Ruleset`

**Review by hburn7 — CHANGES_REQUESTED** [review]

File: API/Entities/Player.cs (hunk 1 above)
Comment (hburn7) [reply API/Entities/Player.cs:46]: And, rename `OsuEnums.Mode` to `OsuEnums.Ruleset`

**Comment by hburn7** [comment]
Comments help us get a little closer towards #129

**Comment by myssto** [comment]
When refactoring `OsuEnums.Mode` to `OsuEnums.Ruleset`, I opted to let Rider change whatever string versions of "mode" it found as well. We can comb through and manually change the rest in a dedicated PR.

File: API/BackgroundWorkers/OsuPlayerDataWorker.cs [hunk 3]
```diff
@@ -39,57 +38,63 @@ private async Task BackgroundTask(CancellationToken cancellationToken)
...
-                    await Task.Delay(UPDATE_INTERVAL_SECONDS * 1000, cancellationToken);
+                    await Task.Delay(UpdateIntervalSeconds * 1000, cancellationToken);
                     continue;
                 }
 
                 foreach (Player? player in playersToUpdate)
                 {
-                    // Fetch ranks for all 4 game modes and update accordingly.
-                    foreach (OsuEnums.Mode gameModeEnum in Enum.GetValues<OsuEnums.Mode>())
+                    // Fetch data for all game modes and update accordingly
```
Comment (hburn7) [inline API/BackgroundWorkers/OsuPlayerDataWorker.cs:54]:
```suggestion
// Fetch data for all rulesets and update accordingly
```

File: API/Osu/Multiplayer/OsuApiService.cs [hunk 4]
```diff
@@ -19,6 +19,9 @@ public class OsuApiUser
 
     [JsonProperty("country")]
     public string Country { get; set; } = null!;
+
+    [JsonProperty("ruleset")]
```
Comment (hburn7) [inline API/Osu/Multiplayer/OsuApiService.cs:23]: I think we can actually remove all of the JsonProperty fields in this class, I don't see them being used anywhere.

**Review by hburn7 — APPROVED** [review]
Looks good

File: API/Osu/Multiplayer/OsuApiService.cs (hunk 4 above)
Comment (myssto) [reply API/Osu/Multiplayer/OsuApiService.cs:23]: I was wondering about those, didn't understand what they were used for so I wasn't going to touch them lol

**Pull request merged**